Reader post sharing / Press this - replace created post content with an embed of the original

Addison-Stavlo opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Resolves item 3 of pe7F0s-P6-p2

## Proposed Changes

* This replaces the content of the post that is started from sharing a post from the reader/press-this with an embed of the original post. 
After:
<img width="500" alt="Screenshot 2023-05-10 at 11 43 51 AM" src="https://github.com/Automattic/wp-calypso/assets/28742426/535e5856-b1e7-4d75-91f5-fb629ce1b9aa">

## Notes on the Embed block

(EDIT- these issues have all been resolved) Working on this seems to be highlighting some issues with the core embed block for WP sites. Fixing these is out of scope of this PR, but seems like they may be a blocker for making these changes. Here are 3 examples of using the embed block in the editor and how it is rendered on the front end:

### In the editor:
<img width="500" alt="Screenshot 2023-05-10 at 12 12 25 PM" src="https://github.com/Automattic/wp-calypso/assets/28742426/d38fada6-d8c8-4ca1-af0f-de8c81420c7f">

* The first and third embed blocks are cut-off
* The second (middle) embed block only renders a link.

### On the front-end:
<img width="500" alt="Screenshot 2023-05-10 at 12 09 50 PM" src="https://github.com/Automattic/wp-calypso/assets/28742426/f04890b7-9902-48c5-8f51-e124d36ec41d">

* The first and third embed blocks are still cut-off, however the third is less-so than before. Buttons for comments and sharing are still cut off on the first.
* The second (middle) that was only a link in the editor ends up rendering as expected on the front-end.

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Build and sync this branch of `wpcom-block-editor` to your sandbox.  ( PCYsg-l4k-p2 for info on this step for wpcom-block-editor )
* sandbox the public-api, widgets, and site you will share the post from.
* Build this calypso branch (or similarly use the calypso.live link supplied below).
* Go to the reader.
* Find a post including a featured image.
* Use the sharing (curved arrow) icon to open the sharing popover.
* select the site you would like to share this from.
* <img width="300" alt="Screenshot 2023-05-09 at 1 40 49 PM" src="https://github.com/Automattic/wp-calypso/assets/28742426/42a9afd4-6e62-41eb-9bc8-80f15af677c6">
* when the editor loads, verify the only post content is an embed of the original post.

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

Addison-Stavlo committed:
replace press this content with an embed of the original post

roo2 commented:
can confirm that the embed block looks pretty poor in this context.
For the post I reblogged, when I publish the page, it basically just renders as a link

shared: https://traveltheworldwithlouise.com/2023/05/10/brae-park-fort-langley/
how my post looks: pbNmQQ-1Bf-p2
In the editor:
<img width="581" alt="Screen Shot 2023-05-11 at 2 12 45 pm" src="https://github.com/Automattic/wp-calypso/assets/22446385/3a379824-67d6-481b-8e82-05daf12bfe4f">

I don't think we should merge unless we can make some improvements to the embed block

roo2 commented on apps/wpcom-block-editor/src/default/features/press-this.js:
```diff
@@ -3,40 +3,14 @@ import { dispatch } from '@wordpress/data';
-		}
-
-		dispatch( 'core/editor' ).resetEditorBlocks( blocks );
+		dispatch( 'core/editor' ).resetEditorBlocks( [ createBlock( 'core/embed', { url: url } ) ] );
```
I think we can improve the appearance by passing some more attributes to the core embed block https://developer.wordpress.org/block-editor/reference-guides/core-blocks/#embed

Addison-Stavlo commented:
> can confirm that the embed block looks pretty poor in this context.
For the post I reblogged, when I publish the page, it basically just renders as a link

Yeah, that seemes similar to #2 in the PR description.  In the editor it just renders a link, but on the published front-end seems perfectly fine:

<img width="676" alt="Screenshot 2023-05-12 at 11 53 50 AM" src="https://github.com/Automattic/wp-calypso/assets/28742426/27310c00-f1ad-4d0a-8a0a-f55e050a0233">

> I don't think we should merge unless we can make some improvements to the embed block

Agreed. 

- [ ] Regarding the first issue of the embeds showing up but having the site info and buttons at the button cut off, this seems to have been due to the marketing banner on free sites and should be resolved with D110647-code
- [ ] For the issue where the editor preview ends up being only a link while the embed shows up on the front-end, will still need to investigate.

Addison-Stavlo committed:
add caption and wp-embed type

roo2 commented:
I tested again, for me it was still showing just a normal link, I looked into it and added a new embed block to the post, the one I added manually had `type=wp-embed` but the one added by the Press This button was `type=link`

Addison-Stavlo commented:
>  I looked into it and added a new embed block to the post, the one I added manually had type=wp-embed but the one added by the Press This button was type=link. 

With the last commit in [add caption and wp-embed type](https://github.com/Automattic/wp-calypso/pull/76776/commits/538a43d2b5517c774dcab72735aecc392f6f7dbe), we started adding them as the `wp-embed` type specifically, but this doesnt seem to help. In all of my testing these sites (seems to be simple sites on custom domains) cause the block to revert from `wp-embed` to `link`. A 3PC issue maybe? not sure yet. - [issue here for refernce](https://github.com/Automattic/wp-calypso/issues/76892)

Addison-Stavlo committed:
Merge branch 'trunk' into update/press-this-to-use-embed

Addison-Stavlo committed:
Merge branch 'trunk' into update/press-this-to-use-embed

Addison-Stavlo committed:
Merge branch 'trunk' into update/press-this-to-use-embed

Addison-Stavlo commented:
Embed block bugs should be resolved across simple/atomic, this is ready for testing again.

davemart-in reviewed (approved):
Looking great! Here's what I saw:

https://github.com/Automattic/wp-calypso/assets/5634774/67c3dadf-08ee-4599-bd76-fa452ab9cfd0

Not sure if we need that last line of text after the embed, but it's not a blocker for launch. 

Nice work pushing through on this one!!

Addison-Stavlo committed:
remove title param and embed caption

Addison-Stavlo commented:
> Not sure if we need that last line of text after the embed, but it's not a blocker for launch.

Updated and removed that caption.

roo2 commented on apps/wpcom-block-editor/src/default/features/press-this.js:
```diff
@@ -3,30 +3,15 @@ import { dispatch } from '@wordpress/data';
-			);
-		}
-
-		dispatch( 'core/editor' ).resetEditorBlocks( blocks );
```
for the few minutes between the calypso change being deployed and the wpcom-block-editor change being deployed, I think embed's won't embed anything, because it relies on `text` being set to embed the link and the calypso part will no longer pass `text` through

Addison-Stavlo replied:
great point!

Addison-Stavlo replied:
Note - i was able to merge this all together and deploy the wpcom-block-editor build before the calypso build got to staging.

Addison-Stavlo replied:
Actually - I forgot about the atomic cache with wpcom-block-editor.  Atomic sites wont pick up the wpcom-block-editor change until tomorrow. For that reason [i removed the calypso side param removals for now ](https://github.com/Automattic/wp-calypso/pull/77451) so atomic can continue to function until they pick up on that update. Come next week we can just revert that removal removal... remove the removals removal... 😆

roo2 reviewed (approved):
LGTM. I think to deploy this you should split the wpcom-block-editor part out and deploy it first

roo2 commented:
Also, this is what happens if the URL cannot be embedded. LGTM 👍  what do you think? I think it happened because the link is `http://` only, not `https://` ?
<img width="763" alt="Screen Shot 2023-05-26 at 11 49 52 am" src="https://github.com/Automattic/wp-calypso/assets/22446385/6567477f-7e39-40df-b115-9372c8e18b2a">

roo2 commented:
Actually I think that is a bug testing on atomic sites! I actually didn't realise this feature supports atomic but it does, try reblogging onto an atomic site. It may be an existing issue unrelated to this PR.
<img width="763" alt="Screen Shot 2023-05-26 at 11 49 52 am" src="https://github.com/Automattic/wp-calypso/assets/22446385/98812a2b-7a09-45a8-a0a4-3bab6076de2a">

roo2 commented:
actually, yes before this diff, the embeds work on Atomic because they won't use embed blocks.
<img width="666" alt="Screen Shot 2023-05-26 at 12 10 58 pm" src="https://github.com/Automattic/wp-calypso/assets/22446385/b6bd0ee1-c9be-452d-8dcc-2eca70a2b1dd">

roo2 reviewed (changes requested):
We'd better fix the atomic issue before merging

Addison-Stavlo commented:
@roo2  what exactly is the issue with Atomic?  We just fixed embedding simple site posts onto atomic yestereday, and embedding atomic onto simple has been working. Was it the insecure URL??

Addison-Stavlo commented:
<img width="864" alt="Screenshot 2023-05-26 at 7 41 38 AM" src="https://github.com/Automattic/wp-calypso/assets/28742426/8f62066d-a883-4cf3-9c4c-00c70be13ce2">

 Reblogging onto an atomic site i see this ^^

Addison-Stavlo commented:
@roo2 - I have been testing this a bit. I can ONLY get what you are showing if the post I am trying to embed is from my WoA dev blog on the dev server pools.  Ive tested a handful of atomic sites in different directions.  I have also tested the link you are trying to embed both on an atomic site from my a8c account as well as a non-a8c account and it worked in both cases (also even with forcing https to http).  I also followed that site and tried sharing from the flow in this branch, and in all tests I am able to embed that post on my sites both simple and atomic:

<img width="827" alt="Screenshot 2023-05-26 at 8 40 19 AM" src="https://github.com/Automattic/wp-calypso/assets/28742426/3097fbca-576a-44b7-a29d-f00892c246c0">

Perhaps this is an environment testing issue? Im tempted to ship and revert or address this separately if there are issues.

roo2 commented:
> I imagine if your atomic dev site is set up to sandbox the public-api, and you have this build patched without having pulled other more recent changes that fixed embedding simple sites on atomic, that you would see this issue.

Oh it could have been something like that, I was testing on an atomic site I'd been using for testing things a long time ago, I might have done something like that. 👍 

Thanks for looking into it
